<launch>
  <!-- simple_planning_simulator parameters -->
  <arg name="loop_rate" default="50.0"/>
  <arg name="simulation_frame_id" default="base_link"/>
  <arg name="map_frame_id" default="map"/>
  <arg name="initialize_source" default="RVIZ"/>
  <arg name="use_waypoints_for_z_position_source" default="true"/>
  <arg name="initial_engage_state" default="true"/>

  <!-- model parameters -->
  <arg name="simulator_model" default="$(find-pkg-share simple_planning_simulator)/config/simple_planning_simulator.param.yaml"/>
  <arg name="vehicle_param_file"/>

  <!-- simple_planning_simulator node -->
  <node pkg="simple_planning_simulator" exec="simple_planning_simulator_exe" name="simple_planning_simulator" output="screen">
    <remap from="/base_trajectory" to="/planning/scenario_planning/trajectory" />
    <remap from="output/current_pose" to="current_pose" />
    <remap from="output/current_twist" to="/vehicle/status/twist" />
    <remap from="output/status" to="/vehicle/status" />
    <remap from="output/control_mode" to="/vehicle/status/control_mode" />
    <remap from="input/vehicle_cmd" to="/control/vehicle_cmd" />
    <remap from="input/turn_signal_cmd" to="/control/turn_signal_cmd" />
    <remap from="input/initial_pose" to="/initialpose" />
    <remap from="input/initial_twist" to="/initialtwist" />
    <remap from="input/engage" to="/vehicle/engage" />

    <param from="$(var simulator_model)"/>
    <param from="$(var vehicle_param_file)"/>

    <param name="loop_rate" value="$(var loop_rate)"/>
    <param name="simulation_frame_id" value="$(var simulation_frame_id)"/>
    <param name="map_frame_id" value="$(var map_frame_id)"/>
    <param name="initialize_source" value="$(var initialize_source)"/>
    <param name="use_waypoints_for_z_position_source" value="$(var use_waypoints_for_z_position_source)"/>
    <param name="initial_engage_state" value="$(var initial_engage_state)"/>
  </node>

  <node pkg="topic_tools" exec="relay" name="twist_relay" output="log" >
    <param name="input_topic" value="/vehicle/status/twist" />
    <param name="output_topic" value="/localization/twist" />
    <param name="type" value="geometry_msgs/msg/TwistStamped" />
  </node>

</launch>
